package/src/bin/omicron-package.rs

```diff
@@ -926,6 +926,62 @@ impl Config {
             _ => bail!("Aborting"),
         }
     }
+
+    /// Returns target packages to be assembled on the builder machine, limited
+    /// to those specified in `only` (if set).
+    // TODO: should this be moved to omicron-zone-package?
```

Comment:
It could, I'm not sure where this line should be drawn, candidly. I think it's "okay" that the logic lives here -- the `omicron-zone-package` is more interested in describing packages than determining how to build them --  but I could also understand wanting that packages to handle more of the details of "figuring out the dependency tree".

I'll defer to your judgement here.